subsys/net/lib/dns/resolve.c

```diff
@@ -74,6 +74,8 @@ NET_BUF_POOL_DEFINE(dns_msg_pool, DNS_RESOLVER_BUF_CTR,
 NET_BUF_POOL_DEFINE(dns_qname_pool, DNS_RESOLVER_BUF_CTR, DNS_MAX_NAME_LEN,
 		    0, NULL);
 
+static K_MUTEX_DEFINE(lock);
```

Comment:
It will work this way, but shouldn't the mutex be actually a part of the `struct dns_resolve_context` structure? There's probably a little use case for such a scenario, but in case there are two DNS contexts running in parallel, a single mutex would be a bottleneck.

Reply:
> shouldn't the mutex be actually a part of the `struct dns_resolve_context` structure

Good point, I will send a new version.